tests/storages_tests/journal_tests/test_journal.py

```diff
@@ -198,3 +230,23 @@ def test_snapshot_given(storage_mode: str, capsys: _pytest.capture.CaptureFixtur
         storage.restore_replay_result(pickle.dumps("hoge"))
         _, err = capsys.readouterr()
         assert err
+
+
+class CustomJournalBackend(BaseJournalLogStorage):
+    def read_logs(self, log_number_from: int) -> list[dict[str, Any]]:
+        return [{"": ""}]
+
+    def append_logs(self, logs: list[dict[str, Any]]) -> None:
+        return
```

Comment:
It would be great to place this definition inside the `test_if_future_warning_occurs`. In addition, the name looks confusing since its name represents it has some features. How about renaming it more precisely such as `InheritDeprecatedClassBackend`?

Reply:
Agreed. I have revised the code.